service/src/test/java/bio/terra/catalog/service/ExternalSystemServiceTest.java

```diff
@@ -0,0 +1,43 @@
+package bio.terra.catalog.service;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.when;
+
+import bio.terra.catalog.common.StorageSystem;
+import bio.terra.catalog.common.StorageSystemInformation;
+import bio.terra.catalog.service.dataset.Dataset;
+import bio.terra.catalog.service.dataset.DatasetAccessLevel;
+import bio.terra.catalog.service.dataset.DatasetDao;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class ExternalSystemServiceTest {
+  @Mock private DatasetDao mockDatasetDao;
+
+  @Test
+  void getDatasets() {
+    String id1 = "storageId1";
```

Comment:
What is the actual storageSourceId for an external dataset? Should it be `''`?

As a follow on, how do we check permissioning for creating external datasets? Can only catalog admins create external datasets (how it is currently in this code)? Or do we want anyone to be able to create external datasets?